### DISCUSSION TITLE
How can I traverse Try<List<String>> using fluent API?

### DISCUSSION THREAD

**Discussion by d3bt3ch** [vavr-io/vavr#2914] [General]
How can I traverse `Try<List<String>>` using fluent API? Please consider the following code snippet.

```
public List<String> foo() {
}

var v = Try.of(() -> foo()) 
        .??
```

I want to traverse the list and do some transformations.

**Comment by pivovarit** [comment]
Try doesn't apply any special treatment to the type of object it holds, so it's fine to iterate it inside one of generic `Try` methods:

```java
Try.of(() -> foo())
  .map(strings -> strings.map(String::toUpperCase))
  .onSuccess(strings -> {
      strings.forEach(System.out::println);
  });
```

**Reply by d3bt3ch** [reply]
I would like to flatten the list so that multiple transformations can be applied. Can it be converted to a stream?

**Reply by pivovarit** [reply]
Not sure I'm following, you want to have `List<Try<String>>` instead of `Try<List<String>>`?

**Reply by d3bt3ch** [reply]
Precisely, yes.

**Reply by pivovarit** [reply]
Then you need to simply isolate each operation into a separate `Try`:

```java
foo().map(s -> Try.of(() -> ...))
```

**Reply by d3bt3ch** [reply]
`foo()` result also needs to be a `Try`
